src/config/config.rs

```diff
@@ -408,7 +428,7 @@ pub struct GraphQL {
   pub name: String,
   #[serde(default, skip_serializing_if = "is_default")]
   pub args: Option<KeyValues>,
-  #[serde(rename = "baseURL")]
+  #[serde(rename = "baseURL", default, skip_serializing_if = "is_default")]
```

Comment:
why do we need to change this?

Reply:
is this a valid case, @tusharmath? I thought every config should have a base URL.

Reply:
Base url can be specified in http or upstream or both. We only validate it during blueprint creation. It could also be inferred during composition of configs. 
So someone could logically set the base url to empty at both places and then override it from a different file.